data/config-schema/src/main/resources/net/corda/schema/configuration/crypto/1.0/corda.crypto.json

```diff
@@ -5,437 +5,195 @@
-              }
+            "attemptTimeoutMills": {
+              "description": "Wait period in milliseconds between attempts, should be reasonably large as some operations may be long, e.g. RSA key generation by the SOFT HSM may take a few seconds",
+              "type": "integer",
+              "default": 20000
             }
           },
           "additionalProperties": false
         },
-        "registration": {
-          "description": "Settings for HSM registration messages processor",
+        "categories": {
+          "description": "Categories which the HSM can be used for",
+          "type": "array",
+          "minItems": 1,
+          "items": {
+            "type": "object",
+            "properties": {
+              "category": {
+                "description": "Category name, could be the wildcard * meaning all which has to be the last item if used",
+                "type": "string"
+              },
+              "policy": {
+                "description": "Defines how private keys are generated for that category, ALIASED - stored in the HSM, WRAPPED - wrapped by the HSM, BOTH - decision is taken by the HSM at runtime",
+                "enum": [
+                  "ALIASED",
+                  "WRAPPED",
+                  "BOTH"
+                ]
+              }
+            },
+            "required": [
+              "category",
+              "policy"
+            ],
+            "additionalProperties": false
+          }
+        },
+        "masterKeyPolicy": {
+          "description": "Wrapping key policy, NONE - no wrapping key is required, SHARED - wrapping key shared by all tenants, UNIQUE - wrapping key unique for each tenant",
+          "enum": [
+            "NONE",
+            "SHARED",
+            "UNIQUE"
+          ]
+        },
+        "masterKeyAlias": {
+          "description": "If the masterKeyPolicy is SHARED then this specifies the alias of the shared key and as such is required",
+          "type": "string"
+        },
+        "capacity": {
+          "description": "Number of keys which can be handled by the HSM, -1 means unlimited capacity",
+          "type": "integer"
+        },
+        "supportedSchemes": {
+          "description": "Key schemes that are supported by the HSM",
+          "type": "array",
+          "minItems": 1,
+          "items": {
+            "type": "string"
+          }
+        },
+        "cfg": {
+          "description": "SOFT HSM specific configuration",
           "type": "object",
-          "default": {},
           "properties": {
-            "maxAttempts": {
-              "description": "Maximum attempts to process a message",
-              "type": "integer",
-              "default": 3
-            },
-            "waitBetweenMills": {
-              "description": "Time between attempts in milliseconds, if the number of values is less than attempts then the last item is repeated",
+            "wrappingKeys": {
```

Comment:
Should this be moved up a level? (i.e. replace `cfg`)

Reply:
Yes good catch, that we should definitely do .

Reply:
Done in [#1047](https://github.com/corda/corda-api/pull/1047)